src/Raven.Client/Documents/Identity/AsyncHiLoIdGenerator.cs

```diff
@@ -76,7 +77,18 @@ public RangeValue(long min, long max)
         /// <returns></returns>
         public Task<string> GenerateDocumentIdAsync(object entity)
         {
-            return NextIdAsync().ContinueWith(task => GetDocumentIdFromId(task.Result));
+            return NextIdAsync().ContinueWith(task =>
```

Comment:
What is the relation between this failing test and this fix?
Would it be better to do await NextIdAsync and then execute second operation?

Reply:
This threw `AggregatedException` instead of `DatabaseDisabledException` as expected.
Before the change the `HiLo` request succeeded while the `SaveChanges` with `DatabaseDisabledException`.

Reply:
So handle that at the caller, or turn this into an `async` method, which will do that too